### PR TITLE
[Xedra Evolved] Maintenance

### LINKED ISSUE #66506: XEDRA Evolved: Inventor Research Error
### Describe the bug

 The following errors happen when you succeed at inventor research for XEDRA Evolved:

DEBUG    : invalid recipe id "software_math_inventor"

 FUNCTION : obj
 FILE     : D:\a\Cataclysm-DDA\Cataclysm-DDA\src\recipe_dictionary.cpp
 LINE     : 67
 VERSION  : b805ca2

and

(continued from above) ERROR : D:\a\Cataclysm-DDA\Cataclysm-DDA\src\recipe_dictionary.cpp:67 [obj] invalid recipe id "software_electronics_reference_inventor"

### Attach save file

[Test-trimmed.tar.gz](https://github.com/CleverRaven/Cataclysm-DDA/files/11892214/Test-trimmed.tar.gz)

### Steps to reproduce

1.  Make a new player with the XEDRA Evolved mod installed.
2. Give yourself an inventor spell item (I use tapped jumper cables)
3. Wait for your character to have an idea (Pass time.)
4. Once you have the basic research recipe do research repeatedly.  This requires a laptop and charges.
5. Error pops up.

### Expected behavior

Error shouldn't pop up.  I'm not sure how the research is supposed to look, as I was exploring it in my main save, but it looks like the errors come from one of the other research recipes that branches off of the basic one.  At least that's what I can tell from a cursory glance at the JSONS.  I think that it might be interfering with those research recipes showing up in game.

### Screenshots

_No response_

### Versions and configuration

- OS: Windows 10
- Game Version: b805ca2
- Graphics Version:  Tiles
- In Game Language:  English
- Mods: DDA, no_npc_food, personal_portal_storms, no_fungal_growth, no_rail_stations, xedra_evolved

### Additional context

```03:32:02.018 ERROR : (error message will follow backtrace)
  #0
    (dbghelp: debug_write_backtrace+0xb5@00007FF6D532A965[cataclysm-tiles.exe+0x5ca965]), 
  #1
    (dbghelp: DebugLog+0x2b8@00007FF6D5329718[cataclysm-tiles.exe+0x5c9718]), 
  #2
    (dbghelp: realDebugmsg+0x1fd@00007FF6D532C9AD[cataclysm-tiles.exe+0x5cc9ad]), 
  #3
    (dbghelp: string_id<recipe>::obj+0xd7@00007FF6D5BDC8B7[cataclysm-tiles.exe+0xe7c8b7]), 
  #4
    (dbghelp: talker_character_const::has_recipe+0x12@00007FF6D639B102[cataclysm-tiles.exe+0x163b102]), 
  #5
    (dbghelp: <lambda_406614664ec528095c9ce0bf66c4d76f>::operator()+0x7a0@00007FF6D5EC9C70[cataclysm-tiles.exe+0x1169c70]), 
  #6
    (dbghelp: talk_effect_t::apply+0x222@00007FF6D5EDD4C2[cataclysm-tiles.exe+0x117d4c2]), 
  #7
    (dbghelp: effect_on_condition::activate+0x13f@00007FF6D59DE3EF[cataclysm-tiles.exe+0xc7e3ef]), 
  #8
    (dbghelp: Character::complete_craft+0xfb6@00007FF6D5D1C1D6[cataclysm-tiles.exe+0xfbc1d6]), 
  #9
    (dbghelp: craft_activity_actor::do_turn+0x571@00007FF6D5C787C1[cataclysm-tiles.exe+0xf187c1]), 
  #10
    (dbghelp: player_activity::do_turn+0x5e0@00007FF6D5804A50[cataclysm-tiles.exe+0xaa4a50]), 
  #11
    (dbghelp: do_turn+0x755@00007FF6D5332305[cataclysm-tiles.exe+0x5d2305]), 
  #12
    (dbghelp: WinMain+0xc75@00007FF6D51BBB25[cataclysm-tiles.exe+0x45bb25]), 
  #13
    (dbghelp: __scrt_common_main_seh+0x106@00007FF6D67B13BA[cataclysm-tiles.exe+0x1a513ba]), 
  #14
    (dbghelp: BaseThreadInitThunk+0x14@00007FFC54D97604[KERNEL32.DLL+0x17604]), 
  #15
    (dbghelp: RtlUserThreadStart+0x21@00007FFC559226A1[ntdll.dll+0x526a1]), 
Backtrace emission took 0 seconds.
(continued from above) ERROR : D:\a\Cataclysm-DDA\Cataclysm-DDA\src\recipe_dictionary.cpp:67 [obj] invalid recipe id "software_math_inventor"
03:34:27.490 ERROR : (error message will follow backtrace)
  #0
    (dbghelp: debug_write_backtrace+0xb5@00007FF6D532A965[cataclysm-tiles.exe+0x5ca965]), 
  #1
    (dbghelp: DebugLog+0x2b8@00007FF6D5329718[cataclysm-tiles.exe+0x5c9718]), 
  #2
    (dbghelp: realDebugmsg+0x1fd@00007FF6D532C9AD[cataclysm-tiles.exe+0x5cc9ad]), 
  #3
    (dbghelp: string_id<recipe>::obj+0xd7@00007FF6D5BDC8B7[cataclysm-tiles.exe+0xe7c8b7]), 
  #4
    (dbghelp: talker_character_const::has_recipe+0x12@00007FF6D639B102[cataclysm-tiles.exe+0x163b102]), 
  #5
    (dbghelp: <lambda_406614664ec528095c9ce0bf66c4d76f>::operator()+0x7a0@00007FF6D5EC9C70[cataclysm-tiles.exe+0x1169c70]), 
  #6
    (dbghelp: talk_effect_t::apply+0x222@00007FF6D5EDD4C2[cataclysm-tiles.exe+0x117d4c2]), 
  #7
    (dbghelp: effect_on_condition::activate+0x13f@00007FF6D59DE3EF[cataclysm-tiles.exe+0xc7e3ef]), 
  #8
    (dbghelp: Character::complete_craft+0xfb6@00007FF6D5D1C1D6[cataclysm-tiles.exe+0xfbc1d6]), 
  #9
    (dbghelp: craft_activity_actor::do_turn+0x571@00007FF6D5C787C1[cataclysm-tiles.exe+0xf187c1]), 
  #10
    (dbghelp: player_activity::do_turn+0x5e0@00007FF6D5804A50[cataclysm-tiles.exe+0xaa4a50]), 
  #11
    (dbghelp: do_turn+0x755@00007FF6D5332305[cataclysm-tiles.exe+0x5d2305]), 
  #12
    (dbghelp: WinMain+0xc75@00007FF6D51BBB25[cataclysm-tiles.exe+0x45bb25]), 
  #13
    (dbghelp: __scrt_common_main_seh+0x106@00007FF6D67B13BA[cataclysm-tiles.exe+0x1a513ba]), 
  #14
    (dbghelp: BaseThreadInitThunk+0x14@00007FFC54D97604[KERNEL32.DLL+0x17604]), 
  #15
    (dbghelp: RtlUserThreadStart+0x21@00007FFC559226A1[ntdll.dll+0x526a1]), 
Backtrace emission took 0 seconds.
(continued from above) ERROR : D:\a\Cataclysm-DDA\Cataclysm-DDA\src\recipe_dictionary.cpp:67 [obj] invalid recipe id "software_electronics_reference_inventor"
03:35:27.360 INFO : Loaded tileset: Chibi_Ultica
03:35:27.360 INFO : Loaded tileset: Chibi_Ultica
03:46:59.964 ERROR : (error message will follow backtrace)
  #0
    (dbghelp: debug_write_backtrace+0xb5@00007FF6D532A965[cataclysm-tiles.exe+0x5ca965]), 
  #1
    (dbghelp: DebugLog+0x2b8@00007FF6D5329718[cataclysm-tiles.exe+0x5c9718]), 
  #2
    (dbghelp: realDebugmsg+0x1fd@00007FF6D532C9AD[cataclysm-tiles.exe+0x5cc9ad]), 
  #3
    (dbghelp: string_id<recipe>::obj+0xd7@00007FF6D5BDC8B7[cataclysm-tiles.exe+0xe7c8b7]), 
  #4
    (dbghelp: talker_character_const::has_recipe+0x12@00007FF6D639B102[cataclysm-tiles.exe+0x163b102]), 
  #5
    (dbghelp: <lambda_406614664ec528095c9ce0bf66c4d76f>::operator()+0x7a0@00007FF6D5EC9C70[cataclysm-tiles.exe+0x1169c70]), 
  #6
    (dbghelp: talk_effect_t::apply+0x222@00007FF6D5EDD4C2[cataclysm-tiles.exe+0x117d4c2]), 
  #7
    (dbghelp: effect_on_condition::activate+0x13f@00007FF6D59DE3EF[cataclysm-tiles.exe+0xc7e3ef]), 
  #8
    (dbghelp: Character::complete_craft+0xfb6@00007FF6D5D1C1D6[cataclysm-tiles.exe+0xfbc1d6]), 
  #9
    (dbghelp: craft_activity_actor::do_turn+0x571@00007FF6D5C787C1[cataclysm-tiles.exe+0xf187c1]), 
  #10
    (dbghelp: player_activity::do_turn+0x5e0@00007FF6D5804A50[cataclysm-tiles.exe+0xaa4a50]), 
  #11
    (dbghelp: do_turn+0x755@00007FF6D5332305[cataclysm-tiles.exe+0x5d2305]), 
  #12
    (dbghelp: WinMain+0xc75@00007FF6D51BBB25[cataclysm-tiles.exe+0x45bb25]), 
  #13
    (dbghelp: __scrt_common_main_seh+0x106@00007FF6D67B13BA[cataclysm-tiles.exe+0x1a513ba]), 
  #14
    (dbghelp: BaseThreadInitThunk+0x14@00007FFC54D97604[KERNEL32.DLL+0x17604]), 
  #15
    (dbghelp: RtlUserThreadStart+0x21@00007FFC559226A1[ntdll.dll+0x526a1]), 
Backtrace emission took 0 seconds.
(continued from above) ERROR : D:\a\Cataclysm-DDA\Cataclysm-DDA\src\recipe_dictionary.cpp:67 [obj] invalid recipe id "software_math_inventor"
03:47:09.855 ERROR : D:\a\Cataclysm-DDA\Cataclysm-DDA\src\recipe_dictionary.cpp:67 [obj] invalid recipe id "software_electronics_reference_inventor"```

### LINKED ISSUE #66723: [Xedra Evolved] Inventor Item Bug Compliation
### Describe the bug

Listed out bugs / issues with inventor items below:

Helmet of Creative Juices:
Material: The helmet is made of various types of protective headwear such as army helmets or firefighter helmets, yet the material is a extremely thin 0.6 milimeter layer of steel + plastic.  If a component in the crafting recipe is a very durable helmet intended to protect the user, the resulting helmet should keep these materials and properties.

Overture:
Description: Likely typo in first couple of words: "Thin piece of close" -> "Thin piece of cloth". Other parts of this item fixes lead from this.
Material: Include cloth in the item materials to match description.
Rigid: The item is largely wires covering a cloth gloves that goes up to the shoulder.  This item therefore shouldn't be rigid.

Vacuum Backpack:
Encumbrance: The backpack reduces the weight of everything placed in it, however it doesn't reduce the item encumbrance.  It follows that if items magically reduce in weight then the amount that a person is encumbered by carrying them should also reduce.  Yet the filled encumbrance of the vacuum backpack is the same as a molle large rucksack which has the same size.

Leg Weight Suspension System:
Material: Made out of an incredibly thin layer of aluminum despite no aluminum in recipe.  As its made up of pipes, a leather belt, and other electronic components, it should take this into account. Leather belt on hips, steel pipes / suspension on legs, etc.
Coverage: Currently covers hips.  However, its a leg suspension system describes as going over the legs.  It should primarily cover the legs (although at a low % coverage as its a suspension system instead of a full coverage piece of armor) along with the belt around the hips.  Could be viewed as being a strapped device.
Rigid: The belt that covers the hips shouldn't be rigid as its a belt.  The leg portion is more complicated, but it is described as being very compact device in the description so its probably not worth conflicting with actual armor.

Force Shield:
Material: Made out of a very thin layer of aluminum despite no aluminum in recipe.  At least probably deserves the protection and material of a normal belt described in description + some section of the actual forcefield device.  Probably worth making strapped too.
Rigid: The force shield is a device strapped to you by a belt.  Therefore it shouldn't be rigid and conflict with actually rigid armor, as its perfectly possible to strap a belt on top of metal pants.

Electronic Halo:
Coverage: Electronic Halo: While it displays information to your eyes, it does so by projecting it onto your eye and the item is described as a crown.  Should be a thin crown worn around the head
Rigid: It is made of rigid materials, but it is described as a small crown with a processor attached.  However, it is a crown with a width of a 10th of a millimeter, so such a close fitting and thin crown really shouldn't conflict with actual rigid headpieces such as an army helmet.

Death Ray:
Recoil: Debatable, but a death ray that shoots an aetheric charge sounds a lot like an energy weapon. Other energy weapons like the A7 laser rifle have no recoil, so the death ray shouldn't either.  Recoil defined in the aetheric_charge AMMO type should be 0.

Inventor's Welder:
Workbenchability: Tools are now able to be plugged into a workbench appliance to receive power / be used nearby.  Currently the inventor's welder cannot be connected at the moment.
Usefullness?: The inventor's welder is a direct copy from the arc welder with the only difference being a wider variety of accepted batteries.  With the ability to plug items into appliances, this does nothing.  The item should have cheaper energy cost or some other benefit to give a reason to make it and a reason why this is mad scientist tech

"Heater":
Workbenchability: Like the inventor's welder, needs to be attachable to a workbench appliance.
Plug in ability: Most tools are able to be plugged into the appliance power grid.  The "Heater" should be as well.

focused energy transformer, incendiary energy transformer, voltaic transformer:
Category: These are weapon mods for the inventor's grenade launcher.  They should be in the misc tab instead of the weapons one.

Beam Scatterer and Focusing Lens (Xedra Versions):
Naming: Despite being different items from the base game Beam Scatterer and Focusing Lens with different effects and ids, they share the exact same name.  Adding even just an "Inventor's " to the start of the name would help differentiate them.

Improved Capacitors:
Balance: These are a slightly nerfed version of the High Density Capacitor from base game.  They both double the energy cost and add 10 fire damage, but the High Density Capacitor also increased range by 5.  For a mad science invention, the mod should at least be slightly better than the default version.

### Attach save file

N/A

### Steps to reproduce

Depends on the item, but most are observable by just looking at the item definitions.

### Expected behavior

Covered in bug description per item

### Screenshots

_No response_

### Versions and configuration

- OS: Windows
    - OS Version: 10.0.19045.3086 (22H2)
- Game Version: 47f8c6e [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Slowdown Fungal Growth [no_fungal_growth],
    No Fungal Monsters [No_Fungi],
    Modularvests [modularvest],
    Aftershock [aftershock],
    Tileset Aftershock [tileset_aftershock],
    <color_cyan>Arcana and Magic Items</color> [Arcana],
    Blaze Industries [blazeindustries],
    Tileset Blaze Industries [tileset_blazeindustries],
    <color_light_green>Cataclysm++</color> [Cata++],
    Magiclysm [magiclysm],
    Tileset Magiclysm [tileset_magiclysm],
    Magiclysm Revamp and Additions [magicrevamp],
    Mana Cores [manacores],
    Xedra Evolved [xedra_evolved],
    Mind Over Matter [mindovermatter],
    Nechronica Redux [Nechronica_Redux],
    Advanced Gear/Technologies [advtech],
    Armor Up! PA Utilities [ArmUpPAUtil],
    Armor Up! Survivor Expansion [ArmUpSurvEx],
    CBM Equipment [CBM_Arms],
    Food preservation via ionizing radiation [irradiate-food],
    Garden Pots [growable-pots],
    Incinerator [incinerator],
    Miscellaneous Magiclysm Expansions [demon_spider_a],
    Pneumatic fun [pneumo_fun],
    Rivtech Beam Swords [beamswordmod],
    XEAS [XEAS],
    <color_light_blue>Lighter Weariness</color> [light_weariness],
    Alternative Map Key [alt_map_key],
    <color_cyan>Arcana</color>/Aftershock Patchmod [Arcana_aftershock_Patch],
    <color_cyan>Arcana</color>/<color_light_green>Cata++</color> Patchmod [Arcana_Cata++_Patch],
    <color_cyan>Arcana</color>/Magiclysm Patchmod [Arcana_Magiclysm_Patch],
    Hydroponics [hydroponics],
    Mining Mod [Mining_Mod],
    Mole Mutation [dig_a_tunna],
    More Survivor Stuff [more_survivor_stuff],
    Mythical Martial Arts [MMA],
    Mythical Martial Arts - Magiclysm [MMA_Magiclysm],
    Translate Complex Dialogue [translate_dialogue],
    Dream of Flight [dream_of_flight],
    Bombastic Perks [bombastic_perks],
    Whaleys TALL Buildings [whaleys_tallbuild],
    sees-player icon, HitButton_iso [sees_player_hitbutton],
    Autodoc Harmless and Perfect [autodoc_harmless_perfect],
    No Bionic Slots [no_cbm_slots],
    Nonperishable Overhaul [Nonperishable_Overhaul],
    Resource Crops [resource_crops],
    Resource Crops for Magiclysm [resource_cropsMagic],
    Sly's Mighty Morphin Mutation Station Reworked [sly_mutation_improvements],
    Stats Through Kills [stats_through_kills],
    Portal Storms Ignore NPCs [personal_portal_storms],
    Minimod: No Skill Rust [minimod_no_rust],
    Minimod: Unpersoned [minimod_unpersoned],
    Targeted Mutations [targeted_mutations],
    Personal Modifications [personal_modifications]
]

### Additional context

Asked to compile list of issues in #66721

### PR DESCRIPTION
**Pull request by GuardianDll**
#### Summary
None
#### Purpose of change
Mod maintenance
fix #66723
fix #66506
#### Describe the solution
See the commit names
#### Additional context
That's pretty shitty fast fixes, because i'm cooking some big stuff, @b3brodie would you like to check please? i think i forgot some stuff with the armor

### CODE REVIEW

File: data/mods/Xedra_Evolved/items/tools.json [hunk 1]
```diff
@@ -63,5 +63,124 @@
...
+    "material": [ "forged_dreamstuff" ],
+    "symbol": ";",
+    "color": "light_gray"
+  },
+  {
+    "id": "dreamforged_ax",
+    "type": "TOOL",
+    "copy-from": "ax",
+    "name": { "str": "dreamforged axe" },
+    "description": "Large, two hand axe to chop some wood.  Light, but still useful",
```
Comment (TheShadowFerret) [inline data/mods/Xedra_Evolved/items/tools.json:178]:
```suggestion
    "description": "Large, two hand axe to chop some wood.  Light, but still useful.",
```
~~Don't axes need weight to chop effectively~~

**Comment by b3brodie** [comment]
The armor changes look pretty good, I don't see anything missed there.  There's just a couple things with the tools and gun mods that need a bit more changed.

Inventor's Welder: Allow attaching to workbench by extending veh_tools_workshop allowed_tools

Heater: Allow attaching to workbench / kitchen unit? by extending the veh_tools_kitchen allowed_tools

Heater: Allow plugging into power grid by adding a plug in action

focused energy transformer, incendiary energy transformer, voltaic transformer: move to misc crafting tab as they are gunmods for the inventor's grenade launcher, not weapons themselves

focusing lens (xedra verison): change name as it shares its name with the basegame focusing lens while being a separate item

Improved Capacitors: needs a balance pass as its just a worse version of the basegame High Density Capacitor

Death Ray: Not sure if this was missed or decided against, but a death ray that consumes aetheric charges to fire sounds a lot like a recoilless laser weapon.  If so, modifying aetheric_charge AMMO to have "recoil": 0 will fix that.

**Comment by b3brodie** [comment]
Also as an aside, since you're adding dreamforged versions of tools a dreamforged scythe would also be a good addition.  A recent pull request added a grass_cutting quality required to harvest some types of crops, so being able to craft a dreamforged one would have some utility.

**Comment by GuardianDll** [comment]
> attaching to workbench

I'm not sure it would work, but also they have a link_up, so you can connect them to the grid directly, i guess it works

> focused energy transformer, incendiary energy transformer, voltaic transformer: move to misc crafting tab as they are gunmods for the inventor's grenade launcher, not weapons themselves

incorrect, they change the gun behavior, allowing to shoot different types of projectiles, similar to hub rifle with swappable barrels, so name them gunmods is kinda incorrect as for me

> focusing lens

i don't think this will exist for long enough to bother myself, also i don't really want to rename or add "inventor" in it

> improved capacitors

same, i don't care about vanilla balance here - if people found one, let them use it

> death ray

nope, that's intensional

> dreamforged scythe

sound odd for me, but i'll think about it

File: data/mods/Xedra_Evolved/items/inventor/armor.json [hunk 2]
```diff
@@ -354,12 +385,22 @@
...
+            { "value": "RANGED_DAMAGE", "multiply": 0.33 },
+            { "value": "WEAPON_DISPERSION", "multiply": 0.33 }
           ]
         }
       ]
     },
-    "armor": [ { "encumbrance": 1, "coverage": 10, "covers": [ "eyes" ], "rigid_layer_only": true } ]
+    "armor": [
+      {
+        "encumbrance_modifiers": [ "NONE" ],
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/items/inventor/armor.json:397]: With the error calling this line but saying it happens at line 421 I'm not sure if this is even an error or if the error is in a different part of the json.

File: data/mods/Xedra_Evolved/items/tools.json [hunk 3]
```diff
@@ -63,5 +63,124 @@
       }
     ],
     "flags": [ "ALLOWS_REMOTE_USE" ]
+  },
+  {
+    "id": "dreamforged_hacksaw",
+    "type": "TOOL",
+    "copy-from": "hacksaw",
+    "name": { "str": "dreamforged hacksaw" },
+    "description": "A sturdy saw, useful for cutting through metal objects. Forged from dreamdross, which makes it much lighter.",
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/items/tools.json:72]:
```suggestion
    "description": "A sturdy saw, useful for cutting through metal objects.  Forged from dreamdross, which makes it much lighter.",
```

File: data/mods/Xedra_Evolved/items/inventor/armor.json (hunk 2 above)
Comment (GuardianDll) [inline data/mods/Xedra_Evolved/items/inventor/armor.json:397]:
```suggestion
```

**Pull request merged**